indexer/Indexer.cc

```diff
@@ -356,6 +358,112 @@ void TuIndexer::saveNamespaceDecl(const clang::NamespaceDecl *namespaceDecl) {
   doc.symbolInfos.insert({symbol, scip::SymbolInformation{}});
 }
 
+void TuIndexer::saveNestedNameSpecifier(
+    const clang::NestedNameSpecifierLoc &argNameSpecLoc) {
+  clang::NestedNameSpecifierLoc nameSpecLoc = argNameSpecLoc;
+
+  auto tryEmit = [this](clang::NestedNameSpecifierLoc nameSpecLoc,
+                        std::optional<std::string_view> optSymbol) {
+    if (!optSymbol.has_value()) {
+      return;
+    }
+    scip::Occurrence occ;
+    auto [range, fileId] = FileLocalSourceRange::fromNonEmpty(
+        this->sourceManager, nameSpecLoc.getLocalSourceRange());
+    range.addToOccurrence(occ);
+    std::string_view symbol = optSymbol.value();
+    occ.set_symbol(symbol.data(), symbol.size());
+    auto &doc = this->documentMap[{fileId}];
+    doc.occurrences.emplace_back(scip::OccurrenceExt{std::move(occ)});
+  };
+
+  while (nameSpecLoc.hasQualifier()) {
+    auto nameSpec = nameSpecLoc.getNestedNameSpecifier();
+    using Kind = clang::NestedNameSpecifier;
+    switch (nameSpec->getKind()) {
+    case Kind::Namespace: {
+      auto namespaceDecl = nameSpec->getAsNamespace();
+      tryEmit(nameSpecLoc,
+              this->symbolFormatter.getNamespaceSymbol(namespaceDecl));
```

Comment:
Looks like getNamedDeclSymbol also handles namespaces, maybe `tryEmit` should take a `NamedDecl *` instead then.